# AD Service State Monitor Parameters
/**:
  ros__parameters:
    # modules_names: string array
    module_names: [
      "map",
      "sensing",
      "perception",
      "localization",
      "planning",
      "control",
      "vehicle",
      "system"
    ]

    # Topic Config
    ## names: string array
    ## configs: top level
    ### names used to declare parameter group
    #### module: string
    #### timeout[s]: double (0: none)
    #### warn_rate[hz]: double (0: none)
    topic_configs:
      names: [
        "/map/vector_map",
        "/map/pointcloud_map",
        "/perception/obstacle_segmentation/pointcloud",
        "/perception/object_recognition/objects",
        "/initialpose3d",
        "/localization/pose_twist_fusion_filter/pose",
        "/planning/mission_planning/route",
        "/planning/scenario_planning/trajectory",
        "/control/trajectory_follower/control_cmd",
        "/control/command/control_cmd",
        "/vehicle/status/velocity_status",
        "/vehicle/status/steering_status",
        "/system/emergency/control_cmd"
      ]

      configs:
        /map/vector_map:
          module: "map"
          timeout: 0.0
          warn_rate: 0.0
          type: "autoware_auto_mapping_msgs/msg/HADMapBin"
          transient_local: True

        /map/pointcloud_map:
          module: "map"
          timeout: 0.0
          warn_rate: 0.0
          type: "sensor_msgs/msg/PointCloud2"
          transient_local: True

        /perception/obstacle_segmentation/pointcloud:
          module: "sensing"
          timeout: 1.0
          warn_rate: 5.0
          type: "sensor_msgs/msg/PointCloud2"
          best_effort: True

        # Can be both with feature array or without
        # In prediction.launch.xml this is set to without
        /perception/object_recognition/objects:
          module: "perception"
          timeout: 1.0
          warn_rate: 5.0
          type: "autoware_auto_perception_msgs/msg/PredictedObjects"
          # This topic could have two different types depending on the launch flags used.
          # The implementation of subscriptions in ROS2 does not allow for multiple types
          # to be defined for a topic. By default this is set to not use have features.
          # type: ["tier4_perception_msgs/msg/DynamicObjectArray", "tier4_perception_msgs/msg/DynamicObjectWithFeatureArray"]

        /initialpose3d:
          module: "localization"
          timeout: 0.0
          warn_rate: 0.0
          type: "geometry_msgs/msg/PoseWithCovarianceStamped"

        /localization/pose_twist_fusion_filter/pose:
          module: "localization"
          timeout: 1.0
          warn_rate: 5.0
          type: "geometry_msgs/msg/PoseStamped"

        /planning/mission_planning/route:
          module: "planning"
          timeout: 0.0
          warn_rate: 0.0
          type: "autoware_auto_planning_msgs/msg/HADMapRoute"
          transient_local: True

        /planning/scenario_planning/trajectory:
          module: "planning"
          timeout: 1.0
          warn_rate: 5.0
          type: "autoware_auto_planning_msgs/msg/Trajectory"

        /control/trajectory_follower/control_cmd:
          module: "control"
          timeout: 1.0
          warn_rate: 5.0
          type: "autoware_auto_control_msgs/msg/AckermannControlCommand"

        /control/command/control_cmd:
          module: "control"
          timeout: 1.0
          warn_rate: 5.0
          type: "autoware_auto_control_msgs/msg/AckermannControlCommand"

        /vehicle/status/velocity_status:
          module: "vehicle"
          timeout: 1.0
          warn_rate: 5.0
          type: "autoware_auto_vehicle_msgs/msg/VelocityReport"

        /vehicle/status/steering_status:
          module: "vehicle"
          timeout: 1.0
          warn_rate: 5.0
          type: "autoware_auto_vehicle_msgs/msg/SteeringReport"

        /system/emergency/control_cmd:
          module: "system"
          timeout: 1.0
          warn_rate: 5.0
          type: "autoware_auto_control_msgs/msg/AckermannControlCommand"

    # Param Config
    ## names: string array
    ## configs: top level
    ### names used to declare parameter group
    #### module: string
    # param_configs:
    #   names: ["/vehicle_info"]
    #   configs:
    #     /vehicle_info:
    #       module: "vehicle"

    # TF Config
    ## names: string array
    ## configs: top level
    ### names used to declare parameter group
    #### module: string
    #### from: string
    #### to: string
    #### timeout[s]: double (0: none)
    tf_configs:
      names: ["map_to_base_link"]
      configs:
        map_to_base_link:
          module: "localization"
          from: "map"
          to: "base_link"
          timeout: 1.0
